src/server/status_server.rs

```diff
@@ -13,6 +17,13 @@ use std::str::FromStr;
 use super::Result;
 use tikv_util::metrics::dump;
 
+#[cfg(not(feature = "no-fail"))]
```

Comment:
Looks like there is too many cfgs, maybe you should extract it to another module to prevent this.

Reply:
"extract it to another module" is a big change or? Perhaps I can do it in another PR? I don't feel I could see clear now how to simply do it :)